dbms/src/Storages/Page/V3/WAL/serialize.cpp

```diff
@@ -332,17 +388,61 @@ String Serializer<PageEntriesEdit>::serializeTo(const PageEntriesEdit & edit)
     return buf.releaseStr();
 };
 
+template <typename PageEntriesEdit>
+String Serializer<PageEntriesEdit>::serializeInCompressedFormTo(const PageEntriesEdit & edit)
+{
+    WriteBufferFromOwnString buf;
+    UInt32 version = WALSerializeVersion::LZ4;
+    writeIntBinary(version, buf);
+    CompressedWriteBuffer compressed_buf(buf);
+    for (const auto & record : edit.getRecords())
+    {
+        switch (record.type)
+        {
+        case EditRecordType::PUT:
+        case EditRecordType::UPSERT:
+        case EditRecordType::VAR_ENTRY:
+        case EditRecordType::UPDATE_REMOTE:
+            serializePutTo(record, compressed_buf);
+            break;
+        case EditRecordType::REF:
+        case EditRecordType::VAR_REF:
+            serializeRefTo(record, compressed_buf);
+            break;
+        case EditRecordType::VAR_DELETE:
+        case EditRecordType::DEL:
+            serializeDelTo(record, compressed_buf);
+            break;
+        case EditRecordType::PUT_EXTERNAL:
+        case EditRecordType::VAR_EXTERNAL:
+            serializePutExternalTo(record, compressed_buf);
+            break;
+        }
+    }
+    compressed_buf.next();
+    return buf.releaseStr();
+}
+
+
 template <typename PageEntriesEdit>
 PageEntriesEdit Serializer<PageEntriesEdit>::deserializeFrom(std::string_view record)
 {
     PageEntriesEdit edit;
     ReadBufferFromMemory buf(record.data(), record.size());
     UInt32 version = 0;
     readIntBinary(version, buf);
-    if (version != 1)
+    CompressedReadBuffer compressed_buf(buf);
+    switch (version)
+    {
+    case WALSerializeVersion::Plain:
+        DB::PS::V3::deserializeFrom(buf, edit);
+        break;
+    case WALSerializeVersion::LZ4:
+        DB::PS::V3::deserializeFrom(compressed_buf, edit);
+        break;
```

Comment:
```suggestion
    switch (version)
    {
    case WALSerializeVersion::Plain:
        DB::PS::V3::deserializeFrom(buf, edit);
        break;
    case WALSerializeVersion::LZ4:
    {
        CompressedReadBuffer compressed_buf(buf);
        DB::PS::V3::deserializeFrom(compressed_buf, edit);
        break;
    }
```

Reply:
It seems we cannot write it in this way because my editor give me an error like "bypass variable initialization".